src/averageAges.js

```diff
@@ -55,7 +65,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon ? people.filter(child => people.some(mom =>
+    mom.name === child.mother) && child.sex === 'm')
+    : people.filter(child =>
+      people.some(mom => mom.name === child.mother));
+
+  const appropriateMom = children.map(child => {
+    const mommy = people.find(person => person.name === child.mother);
+
+    return child.born - mommy.born;
+  }
+  );
+
+  const sumOfDifferences = appropriateMom.reduce((sum, value) => {
+    return sum + value;
+  }, 0);
```

Comment:
The use of `map` here is appropriate as you're transforming the array of children to an array of age differences. However, the `reduce` function inside `sumOfDifferences` could be simplified by removing the curly braces and `return` statement for a more concise expression.